pandas-stubs/core/frame.pyi

```diff
@@ -141,19 +141,19 @@ class _LocIndexerFrame(_LocIndexer):
         self,
         idx: MaskType
         | StrLike
-        | tuple[MaskType | Index | Sequence[Scalar] | Scalar | slice, ...],
+        | tuple[MaskType | Index | Sequence[ScalarT] | Scalar | slice, ...],
         value: S1 | ArrayLike | Series | DataFrame,
     ) -> None: ...
     @overload
     def __setitem__(
         self,
-        idx: tuple[tuple[StrLike | Scalar | slice, ...], StrLike],
+        idx: tuple[tuple[HashableT | Scalar | slice, ...], HashableT],
```

Comment:
This will bind `HashableT` to the same type: if the second tuple element is of type `str`, the first element needs to be a `tuple[str, ...]`.

Reply:
Can you give an example that is valid pandas where we matched before, and don't match now?  Or an example that is valid pandas that should match and we're not catching it?

Reply:
Maybe it doesn't bind within the same annotation - I couldn't create a working example that is not accepted :)

Are the 2nd and 3rd the same?

Reply:
Good catch. That's been there since we pulled the stubs over from Microsoft!

Next commit removes the duplicate.